### ISSUE TITLE
cjxl fails transcoding a jpeg

### ISSUE THREAD

**Issue by andrius4669** [libjxl/libjxl#1810]
**Describe the bug**
cjxl fails encoding 
![v4esrgb](https://user-images.githubusercontent.com/4699695/193827421-49a10436-bde9-4979-babd-fe4770b2def8.jpg)
(sha256sum of the file is 6ed7af5e245da443fcce378340b81db2c5e982d8ae9972e3c2d65f8a5d90f159)
`jpegtran -copy all` doesn't help.
seems like a test image for some color profile stuff but i can't tell if there is something wrong with it why jxl shouldn't be able to encode it.

**To Reproduce**
`cjxl -v -v v4esrgb.jpg v4esrgb.jxl`

**Expected behavior**
Likely shouldn't fail, or at least tell the reason why it does.

**Screenshots**
```
% cjxl -v -v Downloads/failed/v4esrgb.jpg v4esrgb.jxl
JPEG XL encoder v0.7.0 f95da131 [AVX2,SSE4,SSSE3,Unknown]
Read JPEG image with 27972 bytes.
Encoding [Container | JPEG, lossless transcode, effort: 7 | JPEG reconstruction data],
Note: Implicit-default for JPEG is lossless-transcoding. To silence this message, set --lossless_jpeg=(1|0).
JxlEncoderAddJPEGFrame() failed.
```

**Environment**
 - OS: archlinux
 - Compiler version: not sure, distro package
 - CPU type: x86_64
 - cjxl/djxl version string: v0.7.0 f95da131 [AVX2,SSE4,SSSE3,Unknown]

**Comment by andrius4669** [comment]
https://www.color.org/version4html.xalter looks like i've found image source though hashes don't exactly match but that one fails the same way

**Comment by andrius4669** [comment]
looks like Upper_Right.jpg converts fine though so v4 YCC-RGB is fine and only v4 e-sRGB seems to be an issue.

**Comment by jonsneyers** [comment]
```
./lib/jxl/color_encoding_internal.cc:219: JXL_FAILURE: Unable to encode XY -2.443881 0.320882
./lib/jxl/color_encoding_internal.cc:412: JXL_RETURN_IF_ERROR code=1: red_.Set(xy.r)
./lib/jxl/enc_color_management.cc:876: JXL_RETURN_IF_ERROR code=1: IdentifyPrimaries(profile, wp_unadapted, this)
./lib/jxl/jpeg/enc_jpeg_data.cc:311: JXL_RETURN_IF_ERROR code=1: SetColorEncodingFromJpegData(*jpeg_data, &io->metadata.m.color_encoding)
```

Looks like this ICC profile uses primaries that are extremely imaginary. All visible colors have (x,y) coordinates in [0..1], but you can in principle define primaries outside that range and indeed the JPEG XL spec allows setting primaries outside that range (this can be useful if you want to obtain a wider gamut than what is possible by using real primaries only). However, the range that can be represented in JPEG XL is limited to something like [-2,2]. Here the red has a coordinate that is outside that range (x=-2.443881).

There is a workaround for this though: the encoder currently reads the ICC profile and in a case like this where it's a simple space (no lookup table, just primaries + transfer curve), it sets the JPEG XL header to match what is in the ICC profile, if I understand correctly. However in case of JPEG recompression, it also needs to store the actual ICC profile itself, since it needs to reconstruct it exactly (not just semantically equivalently, but bit-exact including comment fields etc). So the JPEG XL header info is redundant in this case, and could in principle be omitted (i.e. replaced by 'Unknown' values or whatever it is we do when we encounter an ICC profile that cannot be represented as an Enum).

So what I'm proposing is instead of failing to transcode in this case, is to only set the details of the Enum space if they can actually be represented by the header syntax. Does that make sense to you, @sboukortt ?

**Comment by VcSaJen** [comment]
[This](https://www.color.org/images/browsertest.jpg) and [this](https://www.color.org/images/browsertest_lut.jpg) from [this color.org page](https://www.color.org/browsertest.xalter) also fails, with same error.
[exiftool output](https://gist.github.com/VcSaJen/6fe23119ae5b583e69a59ca815aca15c)

**Comment by mo271** [comment]
This was fixed in a36aa48a9f56bb744ee7f49e7463d99d53dcea17, please reopen if there still is a problem with these images.
